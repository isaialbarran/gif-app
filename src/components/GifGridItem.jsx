import React from 'react';

const GifGridItem = ({ img }) => {
  const { url, title } = img;

  return (
    <div className='card animate__animated animate__fadeInLeftBig'>
      <img src={url} alt={title} />
    </div>
  );
};

export default GifGridItem;
