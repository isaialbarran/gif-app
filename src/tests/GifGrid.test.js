import React from 'react';
import '@testing-library/jest-dom';

const { shallow } = require('enzyme');
import { GifGrid } from '../components/GifGrid';
import { useFetchGifs } from '../hooks/useFetchGifs';
jest.mock('../hooks/useFetchGifs');

describe('GifGrid test components', () => {
  const category = 'Vegeta';

  test('Match component with snapshot', () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });
    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('should show item when images are loaded', () => {
    const gifs = [
      { id: 'ABC', url: 'https://localhost.jpg', title: 'Cualq cosa' },
      { id: '123', url: 'https://localhost.jpg', title: 'Cualq cosa' },
    ];
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });
    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper.find('p').exists()).toBe(false);
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
  });
});
