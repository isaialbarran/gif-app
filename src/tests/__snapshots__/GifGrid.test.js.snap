// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGrid test components Match component with snapshot 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Vegeta
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    'Cargando....'
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
